### PR TITLE
Switch to gomodules

### PR DESCRIPTION
**Pull request by mattfenwick**

### CODE REVIEW

File: hubapi/component-api.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ type UpgradeGuidance struct {
 	OriginName              string             `json:"originName"`
 	OriginExternalNamespace string             `json:"originExternalNamespace"`
 	OriginExternalId        string             `json:"originExternalId"`
-	VulnerabilityRisk       *VulnerabilityRisk `json:"vulnerabilityRisk, omitempty"`
```
Comment (tandr) [inline hubapi/component-api.go:153]: Ideally this should be a separate PR

File: hubclient/client_test.go [hunk 2]
```diff
@@ -101,7 +101,7 @@ func TestAddClearHeaders(t *testing.T) {
 	assert.Equal(t, "another value", req.Header.Get("Another-Header-Key"))
 
 	// clear the user agent extra header value
-	client.ClearHeaderValue("User-Agent")
```
Comment (tandr) [inline hubclient/client_test.go:104]: How did it compile before?

**Comment by tandr** [comment]
Might want to update readme to specify that it is now uses Go modules, and needs minimum Go's version to run tests.

File: hubclient/client_test.go (hunk 2 above)
Comment (tarunsethi) [reply hubclient/client_test.go:104]: @tandr It didn't. Since its in test code (and hence worked for Code Sight) and probably no one tried to run tests

**Comment by mattfenwick** [comment]
Hi everybody, thanks for the review :) 

I'm going to leave this up for a while, I understand that switching to gomodules could break some things and I certainly don't want that!

If anybody's interested in seeing this merged sooner than later, drop a comment here and we can start the conversation on making this happen!

**Comment by tandr** [comment]
I actually think we should merge it the sooner the better.

**Comment by tandr** [comment]
@mattfenwick I cherry-picked your changes minus going go modules. Don't forget to rebase (or redo) :)
I think once 1.16 is out, this needs to be converted to go modules.

**Comment by mattfenwick** [comment]
Hi @fnxpt @tandr I don't have bandwidth to get this over the finish line , if you want to pick this up feel free!

**Comment by fnxpt** [comment]
Created a new PR.

**Comment by tandr** [comment]
if you have created a new MR, you might as well close this one if it is no longer relevant.

**Comment by fnxpt** [comment]
I think I don't have permissions to close it

**Comment by tandr** [comment]
#78 supersedes this one
